CallCenterSample/Helpers/TextAnalyticsHelper.cs

```diff
@@ -109,48 +123,41 @@ public static async Task<DetectLanguageResult> GetDetectedLanguageAsync(string i
 
             if (!string.IsNullOrEmpty(input))
             {
-                LanguageBatchResult result = await AnalyticsClient.DetectLanguageAsync(new BatchInput(
-                    new List<Input>()
-                    {
-                        new Input("0", input)
-                    }));
-
-                if (result.Documents != null)
-                {
-                    languageResult.Language.Add("iso6391Name", result.Documents[0].DetectedLanguages[0].Iso6391Name);
-                    languageResult.Language.Add("name", result.Documents[0].DetectedLanguages[0].Name);
-                    languageResult.Language.Add("score", result.Documents[0].DetectedLanguages[0].Score.ToString());
-                }
+                DetectLanguageResultCollection result = await AnalyticsClient.DetectLanguageBatchAsync(
+                new List<string>() { input });
 
-                if (result.Errors != null)
+                if (!result[0].HasError)
                 {
-                    // Just return the empty Dictionary
-                }
+                    languageResult.Language.Add("iso6391Name", result[0].PrimaryLanguage.Iso6391Name);
+                    languageResult.Language.Add("name", result[0].PrimaryLanguage.Name);
+                    languageResult.Language.Add("score", result[0].PrimaryLanguage.ConfidenceScore.ToString());
+                }   
             }
 
             return languageResult;
         }
 
         public static async Task<SentimentResult> GetTextSentimentAsync(string input, string language = "en")
         {
-            SentimentResult sentimentResult = new SentimentResult() { Score = 0.5 };
+            SentimentResult sentimentResult = new SentimentResult()
+            {
+                Sentiment = "null",
+                Negative = 0.5,
+                Neutral = 0.5,
+                Positive = 0.5
+            };
 
             if (!string.IsNullOrEmpty(input))
             {
-                SentimentBatchResult result = await AnalyticsClient.SentimentAsync(new MultiLanguageBatchInput(
-                    new List<MultiLanguageInput>()
-                    {
-                        new MultiLanguageInput(language, "0", input)
-                    }));
+                AnalyzeSentimentResultCollection result = await AnalyticsClient.AnalyzeSentimentBatchAsync(
```

Comment:
because there is only 1 input you could just do `DocumentSentiment docSentiment = await AnalyticsClient.AnalyzeSentimentAsync(document);`

Reply:
The return of `AnalyzeSentimentBatchAsync()` has `HasError` property, while I don't see such a property in the return of `AnalyzeSentimentAsync()`.
By `if (!result[0].HasError)`, I'm using it as a `try catch` block that handles errors thrown by `AnalyzeSentimentBatchAsync()`.